File: Array/Easy/1920. Build Array from Permutation.go
package main

import "fmt"

func main() {

	arr := []int{0, 2, 1, 5, 3, 4}
	result := buildArrayPermutation(arr)
	fmt.Println(result)
}

func buildArrayPermutation(arr []int) []int {
	ans := make([]int, len(arr))
	for key := range arr {
		ans[key] = arr[arr[key]]
	}
	return ans
}

File: Miscellaneous/Python/1920. Build Array from Permutation.py
class Solution:
    def buildArray(self, nums):
        result = [0]*len(nums)

        for i in range(len(nums)):
            result[i] = nums[nums[i]]
        
        return result


obj = Solution()
nums = [0,2,1,5,3,4]
ans = obj.buildArray(nums)
print(ans)